docker/gcp_pudl_etl.sh

```diff
@@ -57,38 +52,38 @@ function shutdown_vm() {
     curl -X POST -H "Content-Length: 0" -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://compute.googleapis.com/compute/v1/projects/catalyst-cooperative-pudl/zones/$GCE_INSTANCE_ZONE/instances/$GCE_INSTANCE/stop"
 }
 
-function copy_outputs_to_gcs() {
-    echo "Copying outputs to GCP bucket $PUDL_GCS_OUTPUT"
-    gsutil -m cp -r "$PUDL_OUTPUT" "$PUDL_GCS_OUTPUT"
+function save_outputs_to_gcs() {
+    echo "Copying outputs to GCP bucket $PUDL_GCS_OUTPUT" && \
+    gsutil -m cp -r "$PUDL_OUTPUT" "$PUDL_GCS_OUTPUT" && \
     rm "$PUDL_OUTPUT/success"
 }
 
 function copy_outputs_to_distribution_bucket() {
     # Only attempt to update outputs if we have a real value of BUILD_REF
-    if [ -n "$BUILD_REF" ]; then
-        echo "Removing old $BUILD_REF outputs from GCP distributon bucket."
-        gsutil -m -u "$GCP_BILLING_PROJECT" rm -r "gs://pudl.catalyst.coop/$BUILD_REF"
-        echo "Copying outputs to GCP distribution bucket"
-        gsutil -m -u "$GCP_BILLING_PROJECT" cp -r "$PUDL_OUTPUT/*" "gs://pudl.catalyst.coop/$BUILD_REF"
-
-        echo "Removing old $BUILD_REF outputs from AWS distributon bucket."
-        aws s3 rm "s3://pudl.catalyst.coop/$BUILD_REF" --recursive
-        echo "Copying outputs to AWS distribution bucket"
+    # This avoids accidentally blowing away the whole bucket if it's not set.
+    if [[ -n "$BUILD_REF" ]]; then
+        echo "Removing old $BUILD_REF outputs from GCP distributon bucket." && \
+        gsutil -m -u "$GCP_BILLING_PROJECT" rm -r "gs://pudl.catalyst.coop/$BUILD_REF" && \
+        echo "Copying outputs to GCP distribution bucket" && \
+        gsutil -m -u "$GCP_BILLING_PROJECT" cp -r "$PUDL_OUTPUT/*" "gs://pudl.catalyst.coop/$BUILD_REF" && \
+        echo "Removing old $BUILD_REF outputs from AWS distributon bucket." && \
+        aws s3 rm "s3://pudl.catalyst.coop/$BUILD_REF" --recursive && \
+        echo "Copying outputs to AWS distribution bucket" && \
         aws s3 cp "$PUDL_OUTPUT/" "s3://pudl.catalyst.coop/$BUILD_REF" --recursive
     fi
 }
 
 function zenodo_data_release() {
-    echo "Creating a new PUDL data release on Zenodo."
-    ~/pudl/devtools/zenodo/zenodo_data_release.py --publish --env sandbox --source-dir "$PUDL_OUTPUT"
+    echo "Creating a new PUDL data release on Zenodo." && \
+    ~/pudl/devtools/zenodo/zenodo_data_release.py --publish --env "$1" --source-dir "$PUDL_OUTPUT"
```

Comment:
Added an argument here so we can specify which Zenodo server to release to.

Reply:
Sweet, that's useful! Do we need to pass this in as a `--container-arg` in `build-deploy-pudl.yml`?

Reply:
It's just hard-coded in the main body of the script below right now since we don't yet want to try doing a production release, but I guess when we go to doing production releases, maybe we would like to try and do sandbox releases any time that we're not on a version tag? Just to keep exercising the code?  In which case I guess we could set the value to `sandbox` or `production` depending on which kind of build we're doing in the workflow.  Is that what you were thinking?

Reply:
Yeah, I think eventually setting sandbox/production on every build would be smart. But right now just always pushing to sandbox on dev seems fine. Another improvement for python-script land, I think.